internal/commands/list.go

```diff
@@ -94,38 +121,130 @@ func printHeaders(w io.Writer) {
 	fmt.Fprintln(w, strings.Join(headers, "\t"))
 }
 
-func printValues(w io.Writer, installation *store.Installation) {
+func printValues(w io.Writer, installation Installation) {
 	var values []string
 	for _, column := range listColumns {
 		values = append(values, column.value(installation))
 	}
 	fmt.Fprintln(w, strings.Join(values, "\t"))
 }
 
-func getInstallations(targetContext, configDir string) ([]*store.Installation, error) {
-	appstore, err := store.NewApplicationStore(configDir)
-	if err != nil {
-		return nil, err
-	}
-	installationStore, err := appstore.InstallationStore(targetContext)
-	if err != nil {
-		return nil, err
-	}
+type Installation struct {
+	*store.Installation
+	Services appServices `json:",omitempty"`
+}
+
+func getInstallations(installationStore store.InstallationStore, fetcher ServiceFetcher) ([]Installation, error) {
 	installationNames, err := installationStore.List()
 	if err != nil {
 		return nil, err
 	}
-	installations := make([]*store.Installation, len(installationNames))
+	installations := make([]Installation, len(installationNames))
+	var errs []string
 	for i, name := range installationNames {
 		installation, err := installationStore.Read(name)
 		if err != nil {
 			return nil, err
 		}
-		installations[i] = installation
+		services, err := fetcher.getServices(installation)
+		if err != nil {
+			errs = append(errs, err.Error())
+		}
+		installations[i] = Installation{Installation: installation, Services: services}
 	}
 	// Sort installations with last modified first
 	sort.Slice(installations, func(i, j int) bool {
 		return installations[i].Modified.After(installations[j].Modified)
 	})
-	return installations, nil
+	if len(errs) > 0 {
+		err = errors.New(strings.Join(errs, "\n"))
+	}
+	return installations, err
+}
+
+type ServiceStatus struct {
+	DesiredTasks int
+	RunningTasks int
+}
+
+type appServices map[string]ServiceStatus
+
+type runningService struct {
+	Spec struct {
+		Name string
+	}
+	ServiceStatus ServiceStatus
+}
+
+type serviceFetcher struct {
+	dockerCli command.Cli
+	opts      listOptions
+}
+
+type ServiceFetcher interface {
+	getServices(*store.Installation) (appServices, error)
+}
+
+func (s *serviceFetcher) getServices(installation *store.Installation) (appServices, error) {
+	defer muteDockerCli(s.dockerCli)()
+
+	// bundle without status action returns empty services
+	if !hasAction(installation.Bundle, internal.ActionStatusJSONName) {
+		return nil, nil
+	}
+	creds, err := prepareCredentialSet(installation.Bundle,
+		addDockerCredentials(s.dockerCli.CurrentContext(), s.dockerCli.ContextStore()),
+		addRegistryCredentials(false, s.dockerCli),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var buf bytes.Buffer
+	driverImpl, errBuf, err := cnab.SetupDriver(installation, s.dockerCli, &buf)
```

Comment:
I don't understand why you moved installerContext init out from SetupDriver ? 
Why should the cnab installer _not_ run in desired context when we execute this custom action ?

Reply:
After discussing with @ndeloof I'll move the `installer-context` flag as a global flag.

Reply:
Fixed 👍